weave-js/src/common/util/markdown.ts

```diff
@@ -29,7 +31,43 @@ export function markdownToText(markdown: string) {
   return text;
 }
 
-export function generateHTML(markdown: string) {
+type SanitizationRules = UIConfigOptions['html'];
+export function buildSanitizationSchema(
+  rules: SanitizationRules,
+  schema: Schema = sanitizeRules
+) {
+  /**
+   * These are all the known transformations that can be triggered through
+   * declarative rules passed in from outside weave
+   */
+  const rulesToSchemaMap: Record<SanitizationRules[number], Schema> = {
+    allowScopedStyles: {
+      attributes: {
+        style: ['scoped'],
+      },
+      tagNames: ['style'],
+    },
+  };
+
+  // recursively build up the sanitization object merging in the transformations
+  // coupled to each rule
+  // NOTE: this assumes rules relax strictness. I'm not sure this model would work for
+  // additional strictness on the schema. If we get that scenario we'll need to test
+  const newSchema = rules.reduce(
```

Comment:
given that this is a security related function, I'm in favor of simplifying this code and make it dead obvious what this code does. I think we should hardcode the behavior of each sanitization rule (in this case just the one).  That will likely involve some code copying.

The current code is cleaner than my proposal, but for security stuff I favor it being dead obvious what it does.